### DISCUSSION TITLE
Can't download from my private repo with vcpkg_from_github in GitHub Actions

### DISCUSSION THREAD

**Discussion by bbmedia** [microsoft/vcpkg#35704] [Q&A] [answered]
Hi there
I have a port in my private registry for a private GitHub repo. I try to pass my PAT through an env var (any better ideas?). This works if I test the port locally. But from my GitHub Action, the download (of the correct archive) always returns 404:

```
-- Installing port from location: C:\Users\runneradmin\AppData\Local\vcpkg\registries\git-trees\abc
-- Downloading https://github.com/xyz-AG/xyz/archive/foo.tar.gz -> xyz-AG-xyt-foo.tar.gz...
Downloading https://github.com/xyz-AG/xyz/archive/foo.tar.gz
error: Failed to download from mirror set
error: https://github.com/xyz-AG/xyz/archive/foo.tar.gz: failed: status code 404
```

If I hit the download link in the log-file, the archive gets downloaded successfully. Therefore, the registry and port seems to work ok.

My port looks like:
```
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO xyz-AG/xyz
    REF foo
    SHA512 bar
    AUTHORIZATION_TOKEN "$ENV{MY_PAT}"
    HEAD_REF main
)
vcpkg_cmake_configure(SOURCE_PATH ${SOURCE_PATH})
vcpkg_cmake_install()
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

I pass the PAT as secret in my Action yaml:
```
env:
      MY_PAT: ${{secrets.ACCESS}}
```

What do I miss here? Or is there a better approach to use vcpkg for private repos?

thanks for helping!
Jonas

**Comment by bbmedia** [comment]
Additional info: I tested with a temporary PAT directly added to the `vcpkg_from_github` function in the registry. This works. Therefore, the problem seems to be, on how to pass my PAT from a GitHub secret to `vcpkg_from_github`.

**Reply by Osyotr** [reply, marked as answer]
On windows, environment variables are not passed to ports. You need to add `MY_PAT` to `VCPKG_ENV_PASSTHROUGH_UNTRACKED` https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_env_passthrough_untracked

**Reply by bbmedia** [reply]
Thanks @Osyotr for helping. This did the trick!
